hudi-common/src/main/java/org/apache/hudi/BaseHoodieTableFileIndex.java

```diff
@@ -297,6 +297,16 @@ private void doRefresh() {
                   .orElse(Collections.emptyList())
               )
           );
+    } else if (tableType.equals(HoodieTableType.MERGE_ON_READ) && queryType.equals(HoodieTableQueryType.READ_OPTIMIZED)) {
```

Comment:
This should not be necessary. Let's try to understand why else block doesn't work: `getLatestFileSlices` we're using in there should be filtering base-files produced as a result of pending compaction:
https://github.com/apache/hudi/blob/master/hudi-common/src/main/java/org/apache/hudi/common/table/view/AbstractTableFileSystemView.java#L609

Reply:
@alexeykudinkin 
With `getLatestFileSlices` we can get all latest file-slice in each file-group, but if there is a pending compaction, an empty file-slice will be added to file group in pending compaction, so when `getLatestFileSlices` called, we always get empty file-slice with a pending compaction instant. Actually we want the latest completed file slices in file groups. 
https://github.com/apache/hudi/blob/30e36f35eb2d81515feb6189066418e0fee14f52/hudi-common/src/main/java/org/apache/hudi/common/table/view/AbstractTableFileSystemView.java#L194

Reply:
I see what you're saying now: so the problem is in that empty `FileSlice`, isn't it?

We should fix it differently though: instead we should fix `filterBaseFileAfterPendingCompaction` ([REF](https://github.com/apache/hudi/blob/master/hudi-common/src/main/java/org/apache/hudi/common/table/view/AbstractTableFileSystemView.java#L416)) to filter out file-slice completely if it becomes empty after filtering out the base file. The idea there is that while compaction is running concurrent writer might add new Log files to the `FileSlice` so that's why we're only filtering the Base file, but if FileSlice has no Log files and it becomes empty after filtering the Base file we should filter it out completely.

Reply:
"I see what you're saying now: so the problem is in that empty FileSlice, isn't it?"
yes,  in pending compaction, method `getLatestFileSlice` always got the empty file slice

Reply:
@microbearz are we on the same page regarding the 2d point that i've made? Quoting it below:

> We should fix it differently though: instead we should fix filterBaseFileAfterPendingCompaction ([REF](https://github.com/apache/hudi/blob/master/hudi-common/src/main/java/org/apache/hudi/common/table/view/AbstractTableFileSystemView.java#L416)) to filter out file-slice completely if it becomes empty after filtering out the base file. The idea there is that while compaction is running concurrent writer might add new Log files to the FileSlice so that's why we're only filtering the Base file, but if FileSlice has no Log files and it becomes empty after filtering the Base file we should filter it out completely.